src/freenet/node/updater/NodeUpdateManager.java

```diff
@@ -404,7 +412,7 @@ public void onSuccess(FetchResult result, ClientGetter state) {
 			File temp;
 			FileOutputStream fos = null;
 			try {
-				temp = File.createTempFile(filename, ".tmp", node.getRunDir());
+				temp = File.createTempFile(filename, ".tmp", directory.dir());
```

Comment:
It appears that the idea behind this line was that the temp files are to be
created in the same directory as the files to overwrite, to fail early when
there are permission errors.
Op 25 mrt. 2016 02:29 schreef "Steve Dougherty" <notifications@github.com>:

> In src/freenet/node/updater/NodeUpdateManager.java
> <https://github.com/freenet/fred/pull/516#discussion_r57411151>:
>
> > @@ -404,7 +412,7 @@ public void onSuccess(FetchResult result, ClientGetter state) {
> >  			File temp;
> >  			FileOutputStream fos = null;
> >  			try {
> > -				temp = File.createTempFile(filename, ".tmp", node.getRunDir());
> > +				temp = File.createTempFile(filename, ".tmp", directory.dir());
>
> Is it an intended behavior change to use directories other than the run
> directory for the temporary file? What's the reasoning for that? I was
> going to ask if that would cause problems for packaged versions, but
> packaged versions are updated by the package system anyway so that's not a
> problem.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/freenet/fred/pull/516/files/f7f60d8805c31051c1df24e9f4968f1ebbe46d03#r57411151>
>